src/test/java/seedu/address/model/company/DateTest.java

```diff
@@ -0,0 +1,42 @@
+package seedu.address.model.company;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class DateTest {
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Date(null));
+    }
+    @Test
+    public void constructor_invalidDate_throwsIllegalArgumentException() {
+        String invalidDate = "2103-1-1";
+        assertThrows(IllegalArgumentException.class, () -> new Date(invalidDate));
+    }
+    @Test
+    void isValidDate() {
+        // null date
+        assertThrows(NullPointerException.class, () -> Date.isValidDate(null));
+
+        // invalid date
+        assertFalse(Date.isValidDate("")); // empty string
+        assertFalse(Date.isValidDate(" ")); // spaces only
+        assertFalse(Date.isValidDate("23-12-11")); // invalid Year
+        assertFalse(Date.isValidDate("2023-2-1")); // no leading 0
+        assertFalse(Date.isValidDate("2023-13-01")); // invalid month
+        assertFalse(Date.isValidDate("2023-12-32")); // invalid day
+        assertFalse(Date.isValidDate("phone")); // non-numeric
+
+        // valid phone numbers
```

Comment:
I think you meant this as valid dates

Reply:
This is an invalid date. December should only have 31 days.